File: src/secMot.py
#!/home/ubuntu/vEnvs/rosPy/bin/python

'''
The second motor node. All motor nodes share identical structure but subscribe to
their respective topics and are defined by unique GPIO Pins.
'''

from rospy import loginfo, init_node, Subscriber, spin
from motors.msg import DOFArray
from motorControls import axisMotor

secMot = axisMotor((19, 16), 1, 60, 4.75)

def secAngCall(data):
    '''
    Logs the angle calculated by the master and feeds it to the second motor.
    '''
    loginfo(data.secAng)
    secAng = data.secAng
    secMot.moveToAngle(secAng)


def secondaryMotNode():
    '''
    Subscriber Node for the second motor.
    '''
    init_node('secondaryMotor', anonymous=True)
    Subscriber('motAngs', DOFArray, secAngCall)
    spin()

if __name__ == '__main__':
    secMot.gpioInit()
    secondaryMotNode()

File: src/nodes_cpp/secMot.cpp
#include "ros/ros.h"
#include "motors/DOFArray.h"
#include "AxisMotor/AxisMotor.h"
#include <string>

#include <iostream>
using namespace std;

AxisMotor secMot(19, 16, 1, 60, 4.75);

void callback(const motors::DOFArray::ConstPtr& msg)
{
    float str = msg->secAng;
    ROS_INFO("[%s]\n", to_string(str));
    secMot.moveToAngle(msg->secAng);
}

int main(int argc, char **argv) {

    ros::init(argc, argv, "secMot");
    ros::NodeHandle node;
    ros::Subscriber chat_top_sub = node.subscribe("motAngs", 1000, callback);
    ros::spin();

    return 0;
}